src/main/java/com/liveramp/captain/exception/CaptainNamedException.java

```diff
@@ -0,0 +1,14 @@
+package com.liveramp.captain.exception;
+
+public class CaptainNamedException extends RuntimeException {
+  public CaptainNamedException(String message, Exception e) {
+    super(message, e);
+  }
+
+  public static <ServiceHandle> CaptainNamedException of(Exception e, long id, String stepString, ServiceHandle serviceHandle) {
```

Comment:
Could you explain what the `ServiceHandle` argument is and what one would pass into it when constructing this exception?

Reply:
https://docs.oracle.com/middleware/12211/wls/WLAPI/weblogic/wsee/jws/ServiceHandle.html
Seems like it has to do with JWS service objects but I don't think we use those?

Reply:
Nevermind, I just found this in the captain code https://github.com/LiveRamp/captain/blob/77434c0dc854d0047200f81650b2a7809d9af10c/src/main/java/com/liveramp/captain/waypoint/AsyncWaypoint.java#L19

Reply:
yeah i was just using it from other exceptions. I noticed it was used to pass in some payload etc for logger.

Reply:
i am removing it in this exception. May be we dont need it for this yet.

Reply:
Yeah I think ServiceHandle is used by captain as a generic type to represent a handle that it receives from submitting to a server, which it then tries to persist: https://github.com/LiveRamp/captain/blob/eed7e39bf7271ad5e12cfd5245cc9c573ac8c670/src/main/java/com/liveramp/captain/waypoint/WaypointSubmitterImpl.java#L34

So it makes sense to have that in the persistor exception, but I dont think it makes sense in the submitter exception, since we won't have received a handle from the service if we are throwing this exception